File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    N, M = len(board), len(board[0])
    answer = board[0][0]

    for i in range(1,N):
        for j in range(1,M):
            if board[i][j] == 1:
                board[i][j] = 1+ min(board[i-1][j], board[i-1][j-1], board[i][j-1])
                
                answer = max(board[i][j], answer)

    return answer * answer

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.swift
func solution(_ board: [[Int]]) -> Int {
    let N = board.count
    let M = board[0].count
    var answer = board[0][0]

    var mutableBoard = board

    for i in 1..<N {
        for j in 1..<M {
            if mutableBoard[i][j] == 1 {
                mutableBoard[i][j] = 1 + min(mutableBoard[i - 1][j], mutableBoard[i - 1][j - 1], mutableBoard[i][j - 1])
                answer = max(mutableBoard[i][j], answer)
            }
        }
    }

    return answer * answer
}